### PR TITLE
Task/func tests

### PR DESCRIPTION
**Pull request by kzangeli**
All tests now working with the new implementation of ngsi10 update and query.
But, memory is leaking - one more PR will be necessary to fix this.

### CODE REVIEW

File: test/functionalTest/cases/499_context_providers_basic/update_redirect_delete_operation.test [hunk 1]
```diff
@@ -268,34 +263,49 @@ Date: REGEX(.*)
         </contextAttributeList>
       </contextElement>
       <statusCode>
-        <code>200</code>
-        <reasonPhrase>OK</reasonPhrase>
-        <details>Redirected to context provider localhost:REGEX(\d+)/v1</details>
+        <code>404</code>
```
Comment (fgalan) [inline test/functionalTest/cases/499_context_providers_basic/update_redirect_delete_operation.test]: Diferent from 472 becoming to 404 (or viceversa), which are not big problem, in this case a 200 has become a 404. Is this ok?

File: test/functionalTest/cases/499_context_providers_basic/update_redirect_delete_operation.test [hunk 2]
```diff
@@ -268,34 +263,49 @@ Date: REGEX(.*)
...
 Date: REGEX(.*)
 
 <?xml version="1.0"?>
 <queryContextResponse>
-  <errorCode>
-    <code>404</code>
-    <reasonPhrase>No context element found</reasonPhrase>
-    <details>Redirected to context provider localhost:REGEX(\d+)/v1</details>
-  </errorCode>
+  <contextResponseList>
```
Comment (fgalan) [inline test/functionalTest/cases/499_context_providers_basic/update_redirect_delete_operation.test]: Another remarkable case... this time 404 becomes 200. Is it ok?

File: test/functionalTest/cases/499_context_providers_basic/update_redirect_delete_operation.test (hunk 1 above)
Comment (fgalan) [inline test/functionalTest/cases/499_context_providers_basic/update_redirect_delete_operation.test]: I think this is a (good :) consecuence of fixing #755 in this PR. Thus, NTC.

File: test/functionalTest/cases/499_context_providers_basic/update_redirect_delete_operation.test (hunk 2 above)
Comment (fgalan) [inline test/functionalTest/cases/499_context_providers_basic/update_redirect_delete_operation.test]: I think this is a (good :) consecuence of fixing #755 in this PR. Thus, NTC.

**Comment by fgalan** [comment]
LGTM (but @kzangeli cross-LGTM also required in order to merge the PR).

File: src/lib/mongoBackend/MongoCommonUpdate.cpp [hunk 3]
```diff
@@ -1928,12 +1940,16 @@ void processContextElement(ContextElement*                      ceP,
         /* To finish with this entity processing, search for CPrs in not found attributes and
          * add the corresponding ContextElementResponse to the global response */
         searchContextProviders(tenant, servicePathV, *enP, ceP->contextAttributeVector, cerP);
-        cerP->statusCode.fill(SccOk);
+        if (cerP->statusCode.code == SccNone)
+        {
+          /* It could happend that the StatusCode has already be set, in which case we should
```
Comment (kzangeli) [inline src/lib/mongoBackend/MongoCommonUpdate.cpp]: Not very good English here ... I'll fix it

Comment (kzangeli) [inline src/lib/mongoBackend/MongoCommonUpdate.cpp]: Fixed in 53808fb

**Comment by kzangeli** [comment]
LGTM

**Pull request merged**